tools/ratbagd.py

```diff
@@ -467,6 +472,25 @@ def mapping(self, button):
         return ret
 
     @GObject.Property
+    def macro(self):
+        """A list of (type, keycode) tuples that form the currently set macro,
+        where type is either RatbagdButton.KEY_PRESS or RatbagdButton.KEY_RELEASE
+        and the keycode is as specified in linux/input.h."""
+        return self._get_dbus_property("Macro")
+
+    @macro.setter
+    def macro(self, macro):
+        """Set the macro to the given macro. Note that the type must be one of
+        RatbagdButton.KEY_PRESS or RatbagdButton.KEY_RELEASE and the keycodes
+        must be as specified in linux/input.h.
+
+        @param macro A list of (type, keycode) tuples to form the new macro.
+        """
+        ret = self._dbus_call("SetMacro", "a(uu)", macro)
+        self._set_dbus_property("Macro", "a(uu)", macro, readwrite=False)
```

Comment:
If I'm not mistaken this should also be added to the mapping setters, as yesterday I'd been getting a bunch of [PropertyReadOnly](https://github.com/libratbag/piper/pull/47#issuecomment-317416953) exceptions.